tracer/src/Datadog.Trace/LibDatadog/ServiceDiscovery/ServiceDiscoveryHelper.cs

```diff
@@ -36,7 +36,7 @@ internal static StoreMetadataResult StoreTracerMetadata(TracerSettings tracerSet
                     TracerConstants.Language,
                     TracerConstants.ThreePartVersion,
                     Environment.MachineName,
-                    tracerSettings.ServiceName,
+                    tracerSettings.MutableSettings.DefaultServiceName,
```

Comment:
so this is called only once at start, I'm having trouble understanding what the difference is here if you have any practical examples 😅 
I though that ServiceName was the actual computed service name and that it's what we want, but I'm not sure what could be the difference between calculated / default service name actually 🤯

Reply:
> I though that ServiceName was the actual computed service name and that it's what we want, but I'm not sure what could be the difference between calculated / default service name actually 🤯

Nope, `ServiceName` is _only_ what the customer provides in `DD_SERVICE` 😄  So it's null if they don't set that variable

`DefaultServiceName` is the calculated name that gives a fallback, e.g. `MyApp` if it's `MyApp.exe`

It sounds like `DefaultServiceName` _is_ what you want to me 🙂